src/parse/state/tag.ts

```diff
@@ -376,6 +368,15 @@ function read_attribute(parser: Parser, unique_names: Set<string>) {
 	const colon_index = name.indexOf(':');
 	const type = colon_index !== -1 && get_directive_type(name.slice(0, colon_index));
 
+	if (unique_names.has(name) && type != "EventHandler") {
+		parser.error({
+			code: `duplicate-attribute`,
+			message: 'Attributes need to be unique'
+		}, start);
+	}
+
+	unique_names.add(name);
```

Comment:
It might be nicer to not even add to `unique_names` if we have a `type` of `"EventHandler"`.